src/quo2/components/record_audio/record_audio/view.cljs

```diff
@@ -602,8 +676,91 @@
+                                       seek-time
+                                       #(log/debug "[record-audio] on seek - seek time: " seek-time)
+                                       #(log/error "[record-audio] on seek - error: " %)))
+          :on-value-change          #(when @seeking-audio?
+                                       (reset! audio-current-time-ms %))
+          :thumb-image              (if (colors/dark?) thumb-dark thumb-light)
+          :minimum-track-tint-color (colors/theme-colors colors/primary-50 colors/primary-60)
+          :maximum-track-tint-color (colors/theme-colors
+                                     (if platform/ios? colors/neutral-20 colors/neutral-40)
+                                     (if platform/ios? colors/neutral-80 colors/neutral-60))}]]))])
+
+(defn- time-counter
+  [recording? recording-length-ms ready-to-delete? reviewing-audio? audio-current-time-ms]
+  [:f>
+   (fn []
+     (let [s        (quot (if recording? recording-length-ms audio-current-time-ms) 1000)
+           time-str (gstring/format "%02d:%02d" (quot s 60) (mod s 60))]
+       [rn/view {:style (style/timer-container reviewing-audio?)}
+        (when-not reviewing-audio?
+          [rn/view {:style (style/timer-circle)}])
+        [quo2/text
+         (merge
+          {:size   :label
+           :weight :semi-bold}
+          (when ready-to-delete?
+            {:style (style/timer-text)}))
+         time-str]]))])
+
+(defn- play-button
+  [playing-audio? player-ref playing-timer audio-current-time-ms seeking-audio?]
+  [:f>
+   (fn []
+     [rn/touchable-opacity
+      {:style    (style/play-button)
+       :on-press (fn []
+                   (audio/toggle-playpause-player
+                    @player-ref
+                    #(do
+                       (reset! playing-audio? true)
+                       (reset! playing-timer
+                         (js/setInterval
+                          (fn []
+                            (let [current-time (audio/get-player-current-time @player-ref)
+                                  player-state (audio/get-state @player-ref)
+                                  playing?     (= player-state audio/PLAYING)]
+                              (when (and playing? (not @seeking-audio?) (> current-time 0))
+                                (reset! audio-current-time-ms current-time))))
+                          100)))
+                    #(do
+                       (reset! playing-audio? false)
+                       (when @playing-timer
+                         (js/clearInterval @playing-timer)
+                         (reset! playing-timer nil))
+                       (log/debug "[record-audio] toggle play / pause - success"))
+                    #(log/error "[record-audio] toggle play / pause - error: " %)))}
+      [icons/icon
+       (if @playing-audio? :i/pause :i/play)
+       {:color (colors/theme-colors colors/neutral-100 colors/white)}]])])
```

Comment:
recommend refactoring this into multiple functions

Reply:
Yeah, it's excessively long.

Reply:
Done